staging/src/k8s.io/apimachinery/pkg/api/resource/quantity_test.go

```diff
@@ -736,15 +736,134 @@ func TestQuantityParseEmit(t *testing.T) {
 		in     string
 		expect string
 	}{
+
+		// k (>1)
+		{"1k", "1k"},
+		{"1.23k", "1230"},
+		{"1000k", "1M"},
+		{"1024k", "1024k"},
+		{"1224k", "1224k"},
+		{"1536k", "1536k"},
+		{"1836k", "1836k"},
+
+		// k (<1)
+		{".2k", "200"},
+		{".5k", "500"},
+		{".02k", "20"},
+		{".023k", "23"},
+		{".0234k", "23400m"},
+		{".0004k", "400m"},
+		{".001k", "1"},
+		{".000001k", "1m"},
+		{".000000001k", "1u"},
+		{".000000000001k", "1n"},
+
+		// Ki (>1)
 		{"1Ki", "1Ki"},
+		{"1.23Ki", "1.23Ki"},
+		{"1000Ki", "1000Ki"},
+		{"1024Ki", "1Mi"},
+		{"1224Ki", "1224Ki"},
+		{"1536Ki", "1536Ki"},
+		{"1836Ki", "1836Ki"},
+
+		// Ki (<1)
+		{".2Ki", ".2Ki"},
+		{".5Ki", ".5Ki"},
+		{".02Ki", ".02Ki"},
+		{".023Ki", ".023Ki"},
+		{".0234Ki", ".0234Ki"},
+		{".0004Ki", ".0004Ki"},
+		{".001Ki", ".001Ki"},
```

Comment:
I meant both 1024m and 0.001Ki are correct just different representations in different units.